backend/api/v1/routers/datasets.py

```
@@ -0,0 +1,269 @@
+        logger.error(f"Unexpected error listing datasets: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal server error"
+        )
+
+
+@router.put("/{name}/properties", response_model=Dict[str, str])
+async def update_dataset_properties(
+    name: str,
+    request: UpdateDatasetRequest,
+    service: DatasetManagementService = Depends(get_dataset_service)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)